obitos_convert.py

```diff
@@ -27,20 +28,40 @@ def convert_file(filename):
         )[1]
         row["epidemiological_week_2020"] = brazilian_epidemiological_week(date)[1]
 
+        # There are some missing days on the registral, so default to zero
+        row.update({
+            "new_deaths_sars_2019": 0,
+            "new_deaths_pneumonia_2019": 0,
+            "new_deaths_respiratory_failure_2019": 0,
+            "new_deaths_septicemia_2019": 0,
+            "new_deaths_indeterminate_2019": 0,
+            "new_deaths_others_2019": 0,
+            "new_deaths_sars_2020": 0,
+            "new_deaths_pneumonia_2020": 0,
+            "new_deaths_respiratory_failure_2020": 0,
+            "new_deaths_septicemia_2020": 0,
+            "new_deaths_indeterminate_2020": 0,
+            "new_deaths_others_2020": 0,
+            "new_deaths_covid19": 0
+        })
+
         for item in data:
-            row["new_deaths_sars_2019"] = item.sars_2019
-            row["new_deaths_pneumonia_2019"] = item.pneumonia_2019
-            row["new_deaths_respiratory_failure_2019"] = item.respiratory_failure_2019
-            row["new_deaths_septicemia_2019"] = item.septicemia_2019
-            row["new_deaths_indeterminate_2019"] = item.indeterminate_2019
-            row["new_deaths_others_2019"] = item.others_2019
-            row["new_deaths_sars_2020"] = item.sars_2020
-            row["new_deaths_pneumonia_2020"] = item.pneumonia_2020
-            row["new_deaths_respiratory_failure_2020"] = item.respiratory_failure_2020
-            row["new_deaths_septicemia_2020"] = item.septicemia_2020
-            row["new_deaths_indeterminate_2020"] = item.indeterminate_2020
-            row["new_deaths_others_2020"] = item.others_2020
-            row["new_deaths_covid19"] = item.covid
+            year = item.date.year
+            if year == 2020:
+                row["new_deaths_sars_2020"] = item.sars
+                row["new_deaths_pneumonia_2020"] = item.pneumonia
+                row["new_deaths_respiratory_failure_2020"] = item.respiratory_failure
+                row["new_deaths_septicemia_2020"] = item.septicemia
+                row["new_deaths_indeterminate_2020"] = item.indeterminate
+                row["new_deaths_others_2020"] = item.others
+                row["new_deaths_covid19"] = item.covid
+            elif year == 2019:
+                row["new_deaths_sars_2019"] = item.sars
+                row["new_deaths_pneumonia_2019"] = item.pneumonia
+                row["new_deaths_respiratory_failure_2019"] = item.respiratory_failure
+                row["new_deaths_septicemia_2019"] = item.septicemia
+                row["new_deaths_indeterminate_2019"] = item.indeterminate
+                row["new_deaths_others_2019"] = item.others
```

Comment:
I'd refactor this piece with:

```python
# outside function
from obitos_spider import DeathsSpider

DEATH_CAUSES = list(DeathsSpider.causes_map.keys())

        # [...] inside function
        for item in data:
            year = item.date.year
            for cause in DEATH_CAUSES:
                row[f"new_deaths_{cause}_{year}"] = getattr(item, cause)
```

So we maintain coherence with values expected from the scraper (and delete some lines of code! :)

Reply:
One problem is that covid19 doesn't have the year, should I keep it outside?

Reply:
Nice, I was keeping the code minimal change, but I've refactored this logic to make it less hardcoded.